include/llama.h

```diff
@@ -483,6 +483,13 @@ extern "C" {
     // Get a llama model tensor
     LLAMA_API struct ggml_tensor * llama_get_model_tensor(struct llama_model * model, const char * name);
 
+    // Returns true if the model contains an encoder that requires llama_encode() call
+    LLAMA_API bool llama_model_has_encoder(const struct llama_model * model);
+
+    // For encoder-decoder models, this function returns id of the token that must be provided
+    // to the decoder to start generating output sequence. For other models, it returns -1.
+    LLAMA_API llama_token llama_model_decoder_start_token(const struct llama_model * model);
```

Comment:
Seems better to call it `llama_token_dec()` and move it alongside the other `llama_token_` functions

Reply:
Well I thought about this, and it looks like the other `llama_token_` functions return specific tokens from the vocabulary like BOS, EOS, PAD etc. And decoder start token is not a separate specific token from the vocabulary, but one of the tokens that already existed selected by the model creators to be a token marking the start of the autoregressive decoding process in a given model (it's a PAD token for T5 models). So it's not related to the tokenizer - the tokenizer doesn't know what the decoder start token is, it's a model property. That's why I created a function with a `llama_model_` prefix for this. Also since it's called decoder start token in transformers, I don't think it's a good idea to invent a new new name for this, this will only increase the confusion.

Reply:
@ggerganov Alternatively we could replace it with a more general function that prepares decoder input for seq2seq (encoder-decoder) models instead of simply returning the decoder start token, so it would be something like `llama_model_prepare_decoder_input(llama_model * model, std::vector<llama_token> & seq)`. Or perhaps it's better to explicitly tell the user that this function is only intended for seq2seq (encoder-decoder) models, so maybe `llama_model_seq2seq_prepare_decoder_input`.